### ISSUE TITLE
[UI Feature] Allow for zooming out further and dynamically simplify graph

### ISSUE THREAD

**Issue by LukasBommes** [flyteorg/flyte#3880]
### Motivation: Why do you think this is important?

We implemented a workflow in Flyte that contains so many sub workflows and nodes that the resulting Graph view in the FLyte UI is rather useless. It currently feels like looking at a book page from 2 centimeters distance.

### Goal: What should the final outcome look like, ideally?

 It would help a lot if one could zoom out further to get an overview of the entire graph. 

However, when zooming out too much, the size of elements shown gets too small. It would be great if at a certain zoom level, individual nodes in sub-workflows were hidden and instead a single node (the sub workflow) would be shown.

Basically how it is done in Google Maps: The further you zoom out, the less details are shown.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes
